### PR TITLE
Coverity fix - libYARP_OS #2

### PR DESCRIPTION
**Pull request by damn1**
fix submitted for some other issues raised by Coverity static analyzer libYARP_OS

### CODE REVIEW

File: src/libYARP_OS/include/yarp/os/Random.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #define YARP_OS_RANDOM_H
 
 #include <yarp/os/api.h>
-
+#include <random>
```
Comment (drdanz) [inline src/libYARP_OS/include/yarp/os/Random.h]: Please move this commit on devel

File: src/libYARP_OS/src/NestedContact.cpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ bool NestedContact::fromString(const ConstString& nFullName) {
         nodeName = fullName.substr(0, idx);
         nestedName = fullName.substr(idx+1, fullName.length());
         idx = nestedName.find('/');
-        if (idx==0) return true;
+        if (idx<=0) return true;
```
Comment (drdanz) [inline src/libYARP_OS/src/NestedContact.cpp]: As discussed, this is probably a false positive by coverity, string::find returns a size_t. Please drop this commit

**Comment by drdanz** [comment]
Merged, thanks.

File: src/libYARP_OS/include/yarp/os/Random.h (hunk 1 above)
Comment (traversaro) [reply src/libYARP_OS/include/yarp/os/Random.h]: This commit is also changing a lot the semantics of the call. It deserves a separate PR and  a proper ChangeLog line.

**Pull request merged**